osquery/tables/system/darwin/signature.mm

```diff
@@ -105,11 +111,67 @@ void genSignatureForFile(const std::string& path,
         r["signed"] = INTEGER(1);
         r["identifier"] = stringFromCFString(ident);
 
-        // TODO(andrew-d): can get more information from the signature here?
+        // Get CDHash
+        r["cdhash"] = "";
+        CFTypeRef hashInfo = CFDictionaryGetValue(codeInfo, kSecCodeInfoUnique);
+        if (hashInfo) {
+          r["cdhash"].reserve(CC_SHA1_DIGEST_LENGTH);
+          CFDataRef hash = CFDataRef(hashInfo);
+          // Get the SHA-1 bytes
+          std::stringstream ss;
+          auto bytes = CFDataGetBytePtr(hash);
+          if (bytes) {
+            // Write bytes as hex strings
+            for (unsigned n = 0; n < CC_SHA1_DIGEST_LENGTH; n++) {
+              ss << std::hex << std::setfill('0') << std::setw(2);
+              ss << (unsigned int) bytes[n];
+            }
+            r["cdhash"] = ss.str();
+          }
+          if (r["cdhash"].length() != CC_SHA1_DIGEST_LENGTH * 2) {
+            VLOG(1) << "Error extracting code directory hash";
+            r["cdhash"] = "";
+          }
+        }
+
+        // Team Identifier
+        r["team_identifier"] = "";
+        CFTypeRef teamIdent;
+        if (CFDictionaryGetValueIfPresent(
+                codeInfo, kSecCodeInfoTeamIdentifier, &teamIdent)) {
+          r["team_identifier"] = stringFromCFString((CFStringRef)teamIdent);
+        }
+
+        // Get common name
+        CFArrayRef certChain = (CFArrayRef)CFDictionaryGetValue(
+            codeInfo, kSecCodeInfoCertificates);
+        CFIndex count = CFArrayGetCount(certChain);
```

Comment:
What happens if the value didn't exist and `certChain` is == `nullptr`? The call to `CFArrayGetCount(nullptr)` will fault.